daemon/src/taker_cfd.rs

```diff
@@ -126,14 +141,27 @@ impl Actor {
         let mut conn = self.db.acquire().await?;
         let cfd = load_cfd_by_order_id(order_id, &mut conn).await?;
 
-        let settlement = cfd.calculate_settlement(current_price)?;
+        let proposal = cfd.calculate_settlement(current_price)?;
+
+        if self
+            .current_settlement_proposals
+            .insert(proposal.order_id, proposal.clone())
+            .is_some()
+        {
+            anyhow::bail!(
+                "Settlement proposal for order id {} already present",
+                order_id
+            )
+        }
```

Comment:
This will replace and then error, is that what you want?

Reply:
oops, indeed that would happen! I'll post a quick fix for this one in a sec (bailing if `contains_key()` returns `true)